src/api/environment.cc

```diff
@@ -613,10 +615,14 @@ MaybeLocal<Object> GetPerContextExports(Local<Context> context) {
   if (existing_value->IsObject())
     return handle_scope.Escape(existing_value.As<Object>());
 
+  // To initialize the per-context binding exports, a non-nullptr isolate_data
+  // is needed
+  CHECK(isolate_data);
```

Comment:
Do we need this `CHECK` given that we have it also in `InitializePrivateSymbols`?

Reply:
I think they guard in different levels. `InitializePrivateSymbols` checks that it must be invoked with a non-null `isolate_data`. Here, it checks that if `GetPerContextExports` is invoked the "first time" (like in `node_mksnapshot`), the `isolate_data` must not be null.